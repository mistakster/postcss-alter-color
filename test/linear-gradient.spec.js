const process = require('./utils/process');

const source = `
.i-work {
  background-image: linear-gradient(180deg, rgba(0, 0, 0, 0.5), transparent);
}

.i-dont-work {
  background-image: linear-gradient(180deg, transparent 0, transparent);
}

.i-dont-work-either {
  background-image: linear-gradient(90deg, #009ac1, #42d9ff 37.5%, #009ac1 75%);
}

.i-work-now {
  background-image: linear-gradient(180deg, transparent 0, transparent 0);
}

.i-work-now-too {
  background-image: linear-gradient(
    90deg,
    #009ac1 0,
    #42d9ff 37.5%,
    #009ac1 75%
  );
}
`;

it('should process linear gradients', () => {
  const options = {
    from: '#009ac1',
    to: 'lightgreen',
    preserveAlphaChannel: true
  };

  return process(source, options)
    .andMatchSnapshot();
});
