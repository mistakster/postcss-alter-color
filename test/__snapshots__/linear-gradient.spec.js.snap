// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should process linear gradients 1`] = `
"
.i-work {
  background-image: linear-gradient(180deg,rgba(0,0,0,0.5),transparent);
}

.i-dont-work {
  background-image: linear-gradient(180deg,transparent 0,transparent);
}

.i-dont-work-either {
  background-image: linear-gradient(90deg,lightgreen,#42d9ff 37.5%,lightgreen 75%);
}

.i-work-now {
  background-image: linear-gradient(180deg,transparent 0,transparent 0);
}

.i-work-now-too {
  background-image: linear-gradient(90deg,lightgreen 0,#42d9ff 37.5%,lightgreen 75%);
}
"
`;
